### PR TITLE
Add docs on our roadmap process

### LINKED ISSUE #4233: Add information on process for roadmap
We've formalized how we are prioritizing work, but don't currently have resources for contributors that communicates how to work on features that are on our roadmap.

The simple version is:

* Core team works on issues that are `Accepted`, which implies they are somewhere on our roadmap
* Versioned milestones are our current roadmap
* Named milestones include issues that we aren't yet actively working on. Eventually, these issues will make their way onto a versioned milestone
* Named milestones have priority themselves. We aren't actively working on anything in `Cleanup` or `Refactor` -- these aren't priorities. Our current roadmap today dictates that `Admin UX` features are a high priority.

### PR DESCRIPTION
**Pull request by agjohnson**
This formalized publicly some of our discussions around roadmap and issue
priority.

Fixes #4233

### CODE REVIEW

**Comment by agjohnson** [comment]
cc @RichardLitt 

You might have some input here, let me know if this clears up all your questions or if I missed anything.

File: docs/roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+* New issues coming in will be triaged, but won't yet be considered part of our
+  roadmap.
+* If the issue is a valid bug, it will be assigned the ``Accepted`` label and
+  will be prioritized, likely on an upcoming point release.
+* If the issues is a feature or improvement, the issue might go through a design
+  decision phase before being accepted and assigned to a milestone. This is a
+  good time to discuss how to address the problem technically. Skipping this
+  phase might result in your PR being blocked, sent back to design decision, or
+  perhaps even discarded. It's best to be active here before submitting a PR.
+* The core team will only work on accepted issues, and will only give PR review
```
Comment (RichardLitt) [inline docs/roadmap.rst]: This looks like it discourages drive-by PRs for small items. Should we add a note that if a PR is small enough, you're welcome to file it - we just may not get to it as soon as if you had gone through an issue process?

File: docs/roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+future.
+
+Generally there are 2 or 3 point release milestones lined up. These point
+releases dictate the issues that core team has discussed as priority already.
+Core team should not focus on issues outside these milestones as that implies
+either the issue was not discussed as a priority, or the issue isn't a priority.
+
+Point release milestones try to remain static, but can shift upwards on a
+release that included an unexpected feature addition. Milestones are normally
+realistic and stick to semver, which means that only one feature addition issue
```
Comment (RichardLitt) [inline docs/roadmap.rst]: Might be worth explaining what semver is, or pointing to a description using a link. Same for 'point release'.

File: docs/roadmap.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+these milestones a priorty, they will be moved into point release milestones.
+
+Where to contribute
+~~~~~~~~~~~~~~~~~~~
+
+It's best to pick off an issue from our current point release or group
+milestones, to ensure your pull request gets attention. You can also feel free
+to contribute on our Cleanup or Refactoring milestones. Though not a development
+priority, these issues are generally discrete, easier to jump into than feature
+development, and these milestones are not a place the core team can justify
```
Comment (RichardLitt) [inline docs/roadmap.rst]: and we especially welcome external contributions as these milestones...

File: docs/roadmap.rst (hunk 1 above)
Comment (humitos) [reply docs/roadmap.rst]: Makes sense to me.

File: docs/roadmap.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  perhaps even discarded. It's best to be active here before submitting a PR.
+* The core team will only work on accepted issues, and will only give PR review
+  priority to accepted issues.
+
+Milestones
+~~~~~~~~~~
+
+We maintain two types of milestones: point release milestones for our upcoming
+releases, and group milestones, for blocks of work that have priority in the
+future.
```
Comment (humitos) [inline docs/roadmap.rst]: Do you think that group milestones are good for future priority?

I think it's good to use them as a way to group related amount of work. The priority will be given by the `Accepted` label or when the issue is moved from a group milestone to a point release milestone.

Comment (agjohnson) [reply docs/roadmap.rst]: But we have certain planned features and groups of features that have a priority. For instance in the beginning of the year, this was build stability and admin ux -- and still is. This implies that work in these milestones will be first to be added to our releases, instead of randomly applying issues into our point releases.

Comment (humitos) [reply docs/roadmap.rst]: Yeah, that makes sense to me. I was wondering about when to add to any of this milestones. I suppose that we should add them if just "the topic matches" without considering any kind of priority.

Comment (agjohnson) [reply docs/roadmap.rst]: With some of our work shifting, and with point releases thinning out a little bit, we'll be leaking more issues into point releases now. It's been heavily leaning toward bug fixes for a while.
